src/ListValidator.php

```diff
@@ -18,20 +21,32 @@ class ListValidator extends ValueValidatorObject {
 	 * @since 0.1
 	 *
 	 * @param mixed $value
+	 *
+	 * @throws Exception
 	 */
 	public function doValidation( $value ) {
 		if ( !is_array( $value ) ) {
 			$this->addErrorMessage( 'Not an array' );
 			return;
 		}
 
-		$optionMap = array(
-			'elementcount' => 'range',
-			'maxelements' => 'upperbound',
-			'minelements' => 'lowerbound',
-		);
+		$options = array();
+
+		if ( array_key_exists( 'elementcount', $this->options ) ) {
+			$options['range'] = $this->options['elementcount'];
+		}
+
+		if ( array_key_exists( 'minelements', $this->options ) ) {
+			$options['lowerbound'] = $this->options['minelements'];
+		}
+
+		if ( array_key_exists( 'maxelements', $this->options ) ) {
+			$options['upperbound'] = $this->options['maxelements'];
+		}
```

Comment:
The above could also be `$options = array_intersect_key($this->options, array_flip(['elementcount', …]));`, that's less verbose though.